src/components/GoodsList/GoodsLIst.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const getVisibleGoods = (goods, isReverse, isAlphabetSort, isLengthSort) => {
+  const copyOfGoods = [...goods];
+
+  if (isAlphabetSort) {
+    copyOfGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isLengthSort) {
+    copyOfGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReverse) {
+    copyOfGoods.reverse();
+  }
+
+  return copyOfGoods;
+};
+
+class GoodsList extends React.Component {
+  state = {
+    isReverse: false,
+    isAlphabetSort: false,
+    isLengthSort: false,
+  };
+
+  Reverse = () => {
+    this.setState(state => ({
+      isReverse: !state.isReverse,
+    }));
+  }
+
+  SortByalphabet = () => {
+    this.setState(state => ({
+      isAlphabetSort: !state.isAlphabetSort,
+      isLengthSort: false,
+    }));
+  }
```

Comment:
```suggestion
  SortByalphabet = () => {
    this.setState(state => ({
      isAlphabetSort: !state.isAlphabetSort,
    }));
  }
```
You may have changed only the needed value